README.md

```diff
@@ -1670,7 +1670,80 @@ With all of our builtin logic integrated into the codebase, we can implement any
 
 ##### RangeCheck
 
-TODO
+When working with field elements within the range [0, 2^128), it's important to ensure that these values remain within the specified bounds to prevent unexpected behavior and errors. This check is also used along other operations in many functions, like asserting if a number fits in 250 bits, or exists in other range.  We perform range checks and comparisons using the following approach:
+
+Once we have a RangeCheckRunner that implements the basic builtin interface methods, let's see how this implementation is done:  
+
+We have getters functions just to obtain information about the builtin. The `Name` method is used when iterating through all the builtins of the program so we can switch to the correct execution. 
+
+```go
+func (r *RangeCheckBuiltinRunner) Base() memory.Relocatable {
+	return r.base
+}
+
+func (r *RangeCheckBuiltinRunner) Name() string {
+	return "range_check"
+}
+```
+
+For the `InitializeSegments` we just add a segment to the memory and store in the base attribute, the first adress of the segment. 
+
+```go
+func (r *RangeCheckBuiltinRunner) InitializeSegments(segments *memory.MemorySegmentManager) {
+	r.base = segments.AddSegment()
+}
+```
+
+Next we have the `InitialStack` method, that just returns a stack with the base address appended. 
+
+```go 
+func (r *RangeCheckBuiltinRunner) InitialStack() []memory.MaybeRelocatable {
+	if r.included {
+		stack := []memory.MaybeRelocatable{*memory.NewMaybeRelocatableRelocatable(r.base)}
+		return stack
+	}
+	return []memory.MaybeRelocatable{}
+}
+```
+
+In this case, the `DeduceMemoryCell` is not used in this builtin, so we return nothing. 
+
+```go
+func (r *RangeCheckBuiltinRunner) DeduceMemoryCell(addr memory.Relocatable, mem *memory.Memory) (*memory.MaybeRelocatable, error) {
+	return nil, nil
+}
+```
+
+And finally we have the `AddValidationRule` and the `ValidationRule` methods.
+
+*AddValidationRule* 
+
+Receives the memory and adds to it a new validation rule for the builtin segment. 
+
+*ValidationRule*
+
+Receives the memory and an adress and it checks multiple things. First checks if the value in that address is a `felt` and then if it's inside the range. To do so, it checks that the necessary number of bits for representing the felt is not greater than the bits for representing the upper bound of the range. If it fits in this range, it returns an `Relocatable` array with the address appended. Otherwise returns error. 
```

Comment:
```suggestion
Receives the memory and an address and it checks multiple things. First checks if the value in that address is a `felt` and then if it's inside the range. To do so, it checks that the necessary number of bits for representing the felt is not greater than the bits for representing the upper bound of the range. If it fits in this range, it returns an `Relocatable` array with the address appended. Otherwise returns error. 
```